sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -56,16 +77,69 @@ def test_silhouette():
             assert_almost_equal(score_euclidean, score_dense_with_sampling)
 
 
+def test_silhouette_invalid_block_size():
+    X = [[0], [0], [1]]
+    y = [1, 1, 2]
+    assert_raise_message(ValueError, 'block_size should be at least n_samples '
+                         '* 8 bytes = 1 MiB, got 0',
+                         silhouette_score, X, y, block_size=0)
```

Comment:
That would be identical to just setting it to 64, no? That's tempting, especially because this isn't a learning routine. I don't expect my benchmarks to be optimal, certainly not for any particular platform, or with `n_jobs != 1` (using the default or some other parallel backend); hence my inclination to keep it public. I'll have to chew on this.